asyncapi/twitter/data_types.bal

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Listener related configurations should be included here
+@display {label: "Listener Config"}
+public type ListenerConfig record {
+    @display {label: "API Secret", "description": "Secret specified for Twitter app created"}
+    string apiSecret = DEFAULT_SECRET;
+};
+
+
+# Information about creation event of Tweet/Retweet/Reply/Mention/Quote Tweet 
+#
+# + for_user_id - User ID  
+# + user_has_blocked - Only found in mention events  
+# + tweet_create_events - Description about creation events/activity
+public type TweetEvent record {
+    string for_user_id;
+    boolean? user_has_blocked?;
+    Tweet[] tweet_create_events;
+};
+
+# Information about delete event
+#
+# + for_user_id - User ID  
+# + tweet_delete_events - Description about delete events 
+public type TweetDeleteEvent record {
+    string for_user_id;
+    DeleteEvent[] tweet_delete_events;
+};
+
+# Defines a delete event
+#
+# + status - Information about tweet deleted  
+# + timestamp_ms - Deletion operation timestamp
+public type DeleteEvent record {
+    DeleteStatus? status?;
+    string? timestamp_ms?;
+};
+
+# Describes tweet being deleted
+#
+# + id - ID of tweet deleted
+# + user_id - User ID belong to tweet 
+public type DeleteStatus record {
+    string? id?;
+    string? user_id?;
+};
+
+# Information about follow event
+#
+# + for_user_id - User ID  
+# + follow_events - Description about events when someone follows user
+public type FollowEvent record {
+    string for_user_id;
+    FollowEventInfo[] follow_events;
+};
+
+# Defines a follow event
+#
+# + 'type - Type of event  
+# + created_timestamp - Event created timestamp
+# + target - User object follwed 
+# + 'source - User object which made the follow request 
+public type FollowEventInfo record {
+    string 'type;
+    string? created_timestamp?;
+    User? target?;
+    User? 'source?;
+};
+
+# Information about favorite event
+#
+# + for_user_id - User ID  
+# + favorite_events - Description about event when someone make tweet favorited
+public type FavoriteEvent record {
```

Comment:
```suggestion
public type FavouriteEvent record {
```

Reply:
Since events received from Twitter are with the name `favorite_events` used the same name. WDYT?